### DISCUSSION TITLE
NextAuth session disappearing after page refresh

### DISCUSSION THREAD

**Discussion by shahriyor-sharifjonov** [nextauthjs/next-auth#7567] [Help]
### Question 💬

I'm currently implementing NextAuth sign-in functionality in my application, and overall, things are working fine. However, I've noticed that after refreshing the page, the user.id and customUser data seem to disappear. I would like to ensure that this data persists even after a page refresh. Below is the relevant code snippet that I am using:

import YandexProvider from "next-auth/providers/yandex";
import { MongoDBAdapter } from "@next-auth/mongodb-adapter";
import clientPromise from "@/lib/mongodb";

export const authOptions = {
  providers: [
    YandexProvider({
      clientId: process.env.YANDEX_CLIENT_ID,
      clientSecret: process.env.YANDEX_CLIENT_SECRET,
    }),
  ],
  callbacks: {
    async session({ session, token, user }) {
      const client = await clientPromise;
      const db = client.db();

      const customUsersCollection = db.collection("customUser");

      const existingUser = await customUsersCollection.findOne({ userId: user.id });

      if (!existingUser) {
        const newCustomUser = {
          userId: user.id,
          name: user.name,
          email: user.email,
          image: user.image,
          banner: "",
          balance: 0,
          role: "user",
          pro: 0,
        };

        await customUsersCollection.insertOne(newCustomUser);
        
        if (session?.user) {
          session.user.id = user.id;
          session.customUser = newCustomUser;
        }
      }

      return session;
    },
  },
  adapter: MongoDBAdapter(clientPromise),
};

export default NextAuth(authOptions);
I've taken the necessary steps to set up NextAuth with the MongoDB adapter and implement database sessions. Additionally, I am passing the session data to the SessionProvider in my _app.js file or root component. Despite these efforts, the user.id and customUser information is not retained after refreshing the page.

Could you please provide guidance on how to address this issue and ensure that the user.id and customUser data remains intact even after a page refresh? Thank you in advance for your assistance!

### How to reproduce ☕️

s

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by azeek21** [comment]
Hi, not a professional here, but have you checked that `user` object passed so `session` callback exists after refresh. I don't recall clearly but I had the same problem, if I am not mistaken, It was because `user` object was not always persistent. But I think it was because I was using JWT session not a database session. 

This also may be useful: [Docs on session callback](https://next-auth.js.org/configuration/callbacks#session-callback)
